.github/actions/integration-tests/action.yml

```diff
@@ -79,6 +79,7 @@ runs:
         TEST_BUILD_ID: ${{ inputs.test_build_id }}
         TEST_BUILD_URL: ${{ inputs.test_build_url }}
         TEST_BUILD_REF: ${{ inputs.test_build_ref }}
+        GOMAXPROCS: "4"
```

Comment:
If this doesn't help, maybe reduce parallel tests here: https://github.com/gitpod-io/gitpod/blob/1404189a9ffb85b221a10d43ada38d0f13660196/test/run.sh#L117

Like from 10 to 4?